matchms/similarity/MetadataMatch.py

```diff
@@ -0,0 +1,107 @@
+import logging
+import numpy as np
+from matchms.typing import SpectrumType
+from .BaseSimilarity import BaseSimilarity
+
+
+logger = logging.getLogger("matchms")
+
+
+class MetadataMatch(BaseSimilarity):
+    """Return True if metadata entries of a specified field match between two spectra.
+
+    This is supposed to be used to compare a wide range of possible metadata entries and
+    use this to later select related or similar spectra.
+
+    Example to calculate scores between 2 pairs of spectrums and iterate over the scores
+
+    .. testcode::
+
+        import numpy as np
+        from matchms import calculate_scores
+        from matchms import Spectrum
+        from matchms.similarity import MetadataMatch
+
+        spectrum_1 = Spectrum(mz=np.array([]),
+                              intensities=np.array([]),
+                              metadata={"instrument_type": "orbitrap"})
+        spectrum_2 = Spectrum(mz=np.array([]),
+                              intensities=np.array([]),
+                              metadata={"instrument_type": "qtof"})
+        spectrum_3 = Spectrum(mz=np.array([]),
+                              intensities=np.array([]),
+                              metadata={"instrument_type": "qtof"})
+        spectrum_4 = Spectrum(mz=np.array([]),
+                              intensities=np.array([]),
+                              metadata={"instrument_type": "orbitrap"})
+        references = [spectrum_1, spectrum_2]
+        queries = [spectrum_3, spectrum_4]
+
+        similarity_score = MetadataMatch(field="instrument_type")
+        scores = calculate_scores(references, queries, similarity_score)
+
+        for (reference, query, score) in scores:
+            print(f"Metadata match between {reference.get('id')} and {query.get('id')}" +
+                  f" is {score:.2f}")
+
+    Should output
+
+    .. testoutput::
+
+        Metadata match between 1 and 3 is 0.00
+        Metadata match between 1 and 4 is 1.00
+        Metadata match between 2 and 3 is 1.00
+        Metadata match between 2 and 4 is 0.00
+
+    """
+    # Set key characteristics as class attributes
+    is_commutative = True
+    score_datatype = bool
+
+    def __init__(self, field: str,
+                 matching_type: str = "equal_match",
```

Comment:
Wouldn't the easiest thing here be to actually pass a comparison function which has to evaluate to true and takes two values which are compared?

Reply:
That would indeed work as well.
Only issue I see is that this cannot generally be optimized and will only work for pair-wise comparisons. For the two method given here I could at least do some numby optimization which I just pushed.

We could also do a combination here. So, offer the two default types I sketched and also allow passing a custom function. Would that work?

Reply:
Could we restrict the functions to things which can be parallelized or so? I thought about passing like `==` operator, a `<=` etc. ... but maybe having both options of taking a some keywords which have numba implementations and allowing to pass a lambda with simple parallel for would be good. Btw., has matchms any parallelism, like parallel for loops at the moment?

Reply:
So far parallel computation is depending on the respective scores. By default, no parallel execution will happen. I tried several things, but so far I haven't found a good way to implement parallel execution in Python that will work for all scores and/or all systems. The options which tend to work on all scores/systems can create huge bottlenecks and sometimes make things much slower than without parallel execution (e.g. using Python multiprocessing, or even dask).

Reply:
Okay - I have little experience with parallelism in python, so likely it is better to keep that discussion somewhere else.

Reply:
Yes, we have still the issue #165, but could also update that at some point.